python/pyspark/sql/connect/plan.py

```diff
@@ -721,8 +721,19 @@ def plan(self, session: "SparkConnectClient") -> proto.Relation:
                     rel.join.using_columns.append(self.on)
                 else:
                     rel.join.join_condition.CopyFrom(self.to_attr_or_expression(self.on, session))
-            else:
-                rel.join.using_columns.extend(self.on)
+            elif len(self.on) > 0:
+                if isinstance(self.on[0], str):
+                    rel.join.using_columns.extend(self.on)  # ignore: arg-type
```

Comment:
I am having trouble to make this pass mypy. 

Basically mypy does not buy in `isinstance(self.on[0], str):` is true is `type(self.on) = list[str]`.

Do you know if there is a way to fix it cc @zhengruifeng ?

Reply:
```
python/pyspark/sql/connect/plan.py:726: error: Argument 1 to "extend" of "RepeatedScalarFieldContainer" has incompatible type "Union[List[str], List[Column]]"; expected "Optional[Iterable[str]]"  [arg-type]
```

Reply:
you need to cast:
```suggestion
                    rel.join.using_columns.extend(cast(str, self.on))  # ignore: arg-type
```

Reply:
ah yes it fixes the check. thanks!